app/league/[leagueId]/entry/[entryId]/week/WeekTeams.tsx

```diff
@@ -7,27 +7,69 @@ import { RadioGroup } from '@radix-ui/react-radio-group';
 import { IWeekTeamsProps } from './WeekTeams.interface';
 import { WeeklyPickButton } from '@/components/WeeklyPickButton/WeeklyPickButton';
 
+/**
+ * Formats the date to 'day, mon date' format and the time to either 12 or 24-hour format based on the user's locale.
+ * @param dateStr The date string to format.
+ * @returns The formatted date and time string.
+ */
+const formatDateTime = (dateStr: string): string => {
+  const date = new Date(dateStr);
+
+  const dateOptions: Intl.DateTimeFormatOptions = {
+    weekday: 'short',
+    month: 'short',
+    day: 'numeric',
+  };
+  const timeOptions: Intl.DateTimeFormatOptions = {
+    hour: 'numeric',
+    minute: 'numeric',
+    hour12: true,
+  };
+
+  const formattedDate = date.toLocaleDateString('en-US', dateOptions);
+  const formattedTime = date.toLocaleTimeString('en-US', timeOptions);
+
+  return `${formattedDate} ・ ${formattedTime}`;
+};
+
+/**
+ * Renders the weekly picks page.
+ * @param props The parameters for the weekly picks page.
+ * @param props.field The form field.
+ * @param props.schedule The schedule for the week.
+ * @param props.userPick The user's pick.
+ * @param props.onWeeklyPickChange The function to call when the user's pick changes.
+ * @returns The rendered weekly picks page.
+ */
+import { FormEventHandler } from 'react';
+
 /**
  * Renders the weekly picks page.
  * @param props The parameters for the weekly picks page.
  * @param props.field The form field.
  * @param props.schedule The schedule for the week.
  * @param props.userPick The user's pick.
+ * @param props.onWeeklyPickChange The function to call when the user's pick changes.
  * @returns The rendered weekly picks page.
  */
 const WeekTeams = ({
   field,
   schedule,
   userPick,
+  onWeeklyPickChange,
 }: IWeekTeamsProps): JSX.Element => (
   <>
     {schedule.map((scheduledGame) => (
       <RadioGroup
         onValueChange={field.onChange}
         defaultValue={userPick}
         key={scheduledGame.id}
-        className="grid w-full grid-cols-2 gap-4"
+        className="grid w-full grid-cols-2 gap-4 pb-8"
+        onChange={onWeeklyPickChange as FormEventHandler<HTMLDivElement>}
```

Comment:
No go on assertions. Unless it's a special case, it's most likely your typing that is off. In this case, We need to update IWeekTeamsProps with the correct typing. We used the generic `object` to get by testing.